contrib/mod_ldap.c

```diff
@@ -1257,23 +1257,28 @@ static struct group *pr_ldap_group_lookup(pool *p, char *filter_template,
   return gr;
 }
 
+static void cached_quota_cleanup(void *event_data) {
+  cached_quota = NULL;
+}
+
 static void parse_quota(pool *p, const char *replace, char *str) {
   char **elts, *token;
 
   if (cached_quota == NULL) {
     cached_quota = make_array(p, 9, sizeof(char *));
+    register_cleanup2(p, NULL, cached_quota_cleanup);
   }
 
   elts = (char **) cached_quota->elts;
-  elts[0] = pstrdup(session.pool, replace);
+  elts[0] = pstrdup(p, replace);
```

Comment:
I'm not sure why we were allocating the _contents_ of this structure out of the long-lived `session.pool`, yet using the given pool for the structure memory.  This is both _a)_ confusing, and _b)_ possibly causing small amounts of memory leakage across the lifetime of a single session.